Flush and Periodic Work Requests (#466)
rudderlabs/rudderstack-docs: stream-sources/rudderstack-sdk-integration-guides/rudderstack-android-sdk/flush-events-periodically-in-your-android-application.mdx

---
# slug: "/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-android-sdk/flush-events-periodically-in-your-android-application"
title: "How to Flush Events Periodically in Your Android App"
description: Detailed steps to flush events periodically from the RudderStack SDK’s internal database to Rudder Server.
---

# How to Flush Events Periodically in Your Android App

> RudderStack supports periodic flushing of events from the RudderStack SDK’s internal database to the Rudder server, irrespective of whether the app is opened or not. This ensures that the events triggered just before closing the application are sent from the device even if the application is not opened by the user for a long time.

Reviewer: ```suggestion
RudderStack supports periodic flushing of events from the Android SDK to RudderStack, irrespective of whether your app is opened or not. This ensures that the events that are triggered just before closing the app are sent from the device - even if the app is not opened by the user for a long time.
```

To use this feature, follow these steps:

> 1. Open your `app/build.gradle` and add the WorkManager dependency under `dependencies` as shown below:

Reviewer: What does a WorkManager do? Should we add some kind of a note/context here as we introduce this term for the first time?

```groovy
implementation 'androidx.work:work-runtime:2.7.1'
```

2. Use `withFlushPeriodically()` function while passing the `RudderConfig` object to the `getInstance()` API, as shown below:

```kotlin
rudderClient = RudderClient.getInstance(
  this,

>   WRITE_KEY,

Reviewer: ```suggestion
  <WRITE_KEY>,
```

  RudderConfig.Builder()

>   .withDataPlaneUrl(DATA_PLANE_URL)

Reviewer: ```suggestion
  .withDataPlaneUrl(<DATA_PLANE_URL>)
```

  // enabling periodical flush of events in the DB with a repeat interval of 15 minutes
  .withFlushPeriodically(15, TimeUnit.MINUTES)
  .build()
)
```
## `withFlushPeriodically` Parameters

The `withFlushPeriodically` function takes two parameters:

> * The first parameter is the time interval for the periodic flush worker to be repeated.

Reviewer: ```suggestion
* The first parameter is the number representing the time interval for the flushing to be repeated.
```

> * The second parameter is the time unit of the repeat interval. It can be Minutes, Hours or Days. The minimum value of the repeat interval is 15 minutes.

Reviewer: ```suggestion
* The second parameter is the time unit of the repeat interval. It can be **Minutes**, **Hours**, or **Days**. The minimum value of the repeat interval is **15 minutes**.
```
